src/main/java/org/sagebionetworks/bridge/spring/controllers/CRCController.java

```diff
@@ -349,29 +353,41 @@ private Request addAuthorizationHeader(Request request, Account account) {
         return request;
     }
     
-    private String findIdentifier(Appointment appt, String prefix) {
+    private String findLocation(Appointment appt) {
         if (appt != null && appt.getParticipant() != null) {
             for (AppointmentParticipantComponent component : appt.getParticipant()) {
                 Reference actor = component.getActor();
                 if (actor != null) {
                     String ref = actor.getReference();
-                    if (ref != null && ref.toLowerCase().startsWith(prefix.toLowerCase())) {
-                        return ref.split(prefix)[1];
+                    if (ref != null && ref.toLowerCase().startsWith("location/")) {
+                        return ref.substring(9);
                     }
                 }
             }
         }
-        if (prefix.equals("Patient/")) {
-            throw new BadRequestException("Could not find Bridge user ID.");
-        }
         return null;
     }
+    
+    private String findUserId(Appointment appt) {
+        if (appt != null && appt.getParticipant() != null) {
+            for (AppointmentParticipantComponent component : appt.getParticipant()) {
+                Reference actor = component.getActor();
+                if (actor != null) {
+                    Identifier id = actor.getIdentifier();
+                    if (id != null && USER_ID_VALUE_NS.equals(id.getSystem())) {
+                        return id.getValue();
+                    }
+                }
+            }
+        }
+        throw new BadRequestException("Could not find Bridge user ID.");
+    }
 
-    private String findUserId(Reference ref) {
-        if (ref != null && ref.getReference() != null) {
-            String value = ref.getReference();
-            if (value.toLowerCase().startsWith("patient/")) {
-                return value.substring(8);
+    private String findUserId(Reference subject) {
```

Comment:
Is there a reason the Reference is called a subject here and is called an actor in the method above?

Reply:
Only because the accessor in two of the cases is getSubject() and in the other case is getActor()... despite being labeled as something different, they're both Reference objects.